app/models/course/condition/assessment.rb

```diff
@@ -83,6 +84,11 @@ def validate_unique_dependency
     errors.add(:assessment, :unique_dependency)
   end
 
+  def validate_acyclic_dependency
+    return unless cyclic?
+    errors.add(:assessment, :cyclic_dependency)
```

Comment:
ideally this also tells the user which other conditions make this cyclic, but I think that's hard